src/rules/ResourceICUPluralTranslation.js

```diff
@@ -0,0 +1,249 @@
+            } else if (source[i].type === 8) {
+                sourceTags[source[i].value] = source[i];
+            }
+            if (target[i].type === 5 || target[i].type === 6) {
+                targetPlurals[target[i].value] = target[i];
+            } else if (target[i].type === 8) {
+                targetTags[target[i].value] = target[i];
+            }
+        }
+
+        // for each plural, try to match it up with the target plural by name and check if
+        // there is a translation
+        let results = Object.keys(sourcePlurals).map(name => {
+            const sourcePlural = sourcePlurals[name];
+            const targetPlural = targetPlurals[name];
+            if (!targetPlural) {
+                // missing target plurals are for a different rule, so don't report it here
+                return;
+            }
+            return Object.keys(sourcePlural.options).map(category => {
+                const sourceStr = this.reconstruct(sourcePlural.options[category].value);
+                const targetStr = this.reconstruct(targetPlural.options[category].value);
+                let result = [];
+
+                // use case- and whitespace-insensitive match
+                if (sourceStr.toLowerCase() === targetStr.toLowerCase()) {
+                    let value = {
+                        severity: "warning",
+                        description: `Translation of the category \'${category}\' is the same as the source.`,
+                        rule: this,
+                        id: resource.getKey(),
+                        source: `${category} {${sourceStr}}`,
+                        highlight: `Target: <e0>${category} {${targetStr}}</e0>`,
+                        pathName: file,
+                        locale: resource.getTargetLocale()
+                    };
+                    if (typeof(resource.lineNumber) !== 'undefined') {
+                        value.lineNumber = resource.lineNumber;
+                    }
+                    result.push(new Result(value));
+                }
+
+                // now the plurals may have plurals nested in them, so recursively check them too
+                return result.concat(this.traverse(resource, file, sourcePlural.options[category].value, targetPlural.options[category].value));
+             }).flat().filter(result => result);
+        }).flat().filter(result => result);
+
+        // now recursively handle the tags
+        results = results.concat(Object.keys(sourceTags).map(name => {
+            const sourceTag = sourceTags[name];
+            const targetTag = targetTags[name];
+            if (!targetTag) {
+                // missing target tags are for a different rule, so don't report it here
+                return;
+            }
+            return this.traverse(resource, file, sourceTag.children, targetTag.children);
+        }).flat().filter(result => result));
+        return results.length < 2 ? results[0] : results;
```

Comment:
Duplicate check, also in line 203

Reply:
We need to do flat() in multiple places to keep it as one long array, but the filter only needs to be done at the top level